diag_manager/fms_diag_axis_object.F90

```diff
@@ -390,7 +399,7 @@ function check_if_subaxis_exists(obj,bounds) &
   !> @return Length of the 2D domain
   function get_length(obj, cart_axis, domain_position, global_length) &
```

Comment:
This is not a pure function because `mpp_get_compute_domain` which is used is not.

```
error #7137: Any procedure referenced in a PURE procedure, including one referenced via a defined operation or assignment, must have an explicit interface and be declared PURE.   [MPP_DOMAINS_MOD^MPP_GET_COMPUTE_DOMAIN2D]
      if (trim(cart_axis) == "X") call mpp_get_compute_domain(this%Domain2, xsize=length, position=domain_position)

```